controllers/helpers.go

```diff
@@ -241,6 +241,22 @@ func userAssignedIdentityCloudProviderConfig(d azure.ClusterScoper, identityID s
 	return controlPlaneConfig, workerConfig
 }
 
+// GetExtendedLocationName returns the name for ExtendedLocation.
+func GetExtendedLocationName(d azure.ClusterScoper) string {
+	if d.ExtendedLocation() == nil {
+		return ""
+	}
+	return d.ExtendedLocation().Name
+}
+
+// GetExtendedLocationType returns the type for ExtendedLocation.
+func GetExtendedLocationType(d azure.ClusterScoper) string {
+	if d.ExtendedLocation() == nil {
+		return ""
+	}
+	return d.ExtendedLocation().Type
+}
+
```

Comment:
Can we convert these to be methods of azure.ClusterScoper?

Reply:
Also, move this to azure/scope/cluster.go file. 

Example: 
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/scope/cluster.go#L699

Reply:
Has convert to ExtendedLocationName() and ExtendedLocationType() in azure.ClusterScoper and move to azure/scope/cluster.go file.